### DISCUSSION TITLE
Does DataTable control has a fixed row count of 5?

### DISCUSSION THREAD

**Discussion by kspviswa** [flet-dev/flet#3581] [Q&A] [answered]
### Question

I tried to include 10 DataRow(s) in my DataTable, but only 5 are displayed.

```
    for i in range(1,10):
        print('##### Dumping')
        rows.append(ft.DataRow(cells=[
                ft.DataCell(ft.Text(f'Dump_{i}')),
                ft.DataCell(ft.Text('Dump')),
                ft.DataCell(content = ft.Row([
                                    ft.IconButton(
                        icon=ft.icons.CHECK_CIRCLE_OUTLINE,
                        icon_color="green400",
                        icon_size=20,
                        tooltip="Use this model",
                    ),
                                    ft.IconButton(
                        icon=ft.icons.DELETE_FOREVER_ROUNDED,
                        icon_color="pink600",
                        icon_size=20,
                        tooltip="Delete this model",
                    ),
                ])),
            ]))
```

Tried setting height, min_data_row_height, max_data_row_height but no change in the behaviour. What should be enabled ?

Ideally I'm looking for autoscroll or pagination if number of rows go beyond a configurable number.

### Code sample

```python
for i in range(1,10):
        print('##### Dumping')
        rows.append(ft.DataRow(cells=[
                ft.DataCell(ft.Text(f'Dump_{i}')),
                ft.DataCell(ft.Text('Dump')),
                ft.DataCell(content = ft.Row([
                                    ft.IconButton(
                        icon=ft.icons.CHECK_CIRCLE_OUTLINE,
                        icon_color="green400",
                        icon_size=20,
                        tooltip="Use this model",
                    ),
                                    ft.IconButton(
                        icon=ft.icons.DELETE_FOREVER_ROUNDED,
                        icon_color="pink600",
                        icon_size=20,
                        tooltip="Delete this model",
                    ),
                ])),
            ]))
```

### ------------------------------------------------------

- [X] I have searched for answers to my question both in the issues and in previous discussions.

**Comment by ndonkoHenri** [comment]
Please share some runnable code.

**Comment by burhansvural** [comment, marked as answer]
> ### Question
> I tried to include 10 DataRow(s) in my DataTable, but only 5 are displayed.
> 
> ```
>     for i in range(1,10):
>         print('##### Dumping')
>         rows.append(ft.DataRow(cells=[
>                 ft.DataCell(ft.Text(f'Dump_{i}')),
>                 ft.DataCell(ft.Text('Dump')),
>                 ft.DataCell(content = ft.Row([
>                                     ft.IconButton(
>                         icon=ft.icons.CHECK_CIRCLE_OUTLINE,
>                         icon_color="green400",
>                         icon_size=20,
>                         tooltip="Use this model",
>                     ),
>                                     ft.IconButton(
>                         icon=ft.icons.DELETE_FOREVER_ROUNDED,
>                         icon_color="pink600",
>                         icon_size=20,
>                         tooltip="Delete this model",
>                     ),
>                 ])),
>             ]))
> ```
> 
> Tried setting height, min_data_row_height, max_data_row_height but no change in the behaviour. What should be enabled ?
> 
> Ideally I'm looking for autoscroll or pagination if number of rows go beyond a configurable number.
> 
> ### Code sample
> ```python
> for i in range(1,10):
>         print('##### Dumping')
>         rows.append(ft.DataRow(cells=[
>                 ft.DataCell(ft.Text(f'Dump_{i}')),
>                 ft.DataCell(ft.Text('Dump')),
>                 ft.DataCell(content = ft.Row([
>                                     ft.IconButton(
>                         icon=ft.icons.CHECK_CIRCLE_OUTLINE,
>                         icon_color="green400",
>                         icon_size=20,
>                         tooltip="Use this model",
>                     ),
>                                     ft.IconButton(
>                         icon=ft.icons.DELETE_FOREVER_ROUNDED,
>                         icon_color="pink600",
>                         icon_size=20,
>                         tooltip="Delete this model",
>                     ),
>                 ])),
>             ]))
> ```
> 
> ### Error message
> _No response_
> 
> ### ------------------------------------------------------
> * [x]  I have searched for answers to my question both in the issues and in previous discussions.

As you turn the mouse wheel on the datatable, the data below is displayed.

**Comment by kspviswa** [comment]
Thanks for the tip. I added height to the surrounding ft.Col and set the scroll mode and it all worked
